### PR TITLE
ant for `simple` example

### LINKED ISSUE #429: examples/simple ant run fails with ExecuteJava can not access a member of class Yylex with modifiers "public static"
Steps:
```
cd jflex/examples/simple
ant run
```

Expected:
Execute Yylex

Actual:
> build:
>     [javac] /Users/regis/Projects/jflex/jflex/examples/simple/build.xml:25: warning: 'includeantruntime' was not set, defaulting to build.sysclasspath=last; set to false for repeatable builds
> 
> run:
>      [java] java.lang.IllegalAccessException: Class org.apache.tools.ant.taskdefs.ExecuteJava can not access a member of class Yylex with modifiers "public static"
>      [java] 	at org.apache.tools.ant.taskdefs.ExecuteJava.execute(ExecuteJava.java:194)
>      [java] 	at org.apache.tools.ant.taskdefs.Java.run(Java.java:861)
>      [java] 	at org.apache.tools.ant.taskdefs.Java.executeJava(Java.java:231)
>      [java] 	at org.apache.tools.ant.taskdefs.Java.executeJava(Java.java:135)
>      [java] 	at org.apache.tools.ant.taskdefs.Java.execute(Java.java:108)
>      [java] 	at org.apache.tools.ant.UnknownElement.execute(UnknownElement.java:292)
>      [java] 	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
>      [java] 	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
>      [java] 	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
>      [java] 	at java.lang.reflect.Method.invoke(Method.java:498)
>      [java] 	at org.apache.tools.ant.dispatch.DispatchUtils.execute(DispatchUtils.java:99)
>      [java] 	at org.apache.tools.ant.Task.perform(Task.java:350)
>      [java] 	at org.apache.tools.ant.Target.execute(Target.java:449)
>      [java] 	at org.apache.tools.ant.Target.performTasks(Target.java:470)
>      [java] 	at org.apache.tools.ant.Project.executeSortedTargets(Project.java:1388)
>      [java] 	at org.apache.tools.ant.Project.executeTarget(Project.java:1361)
>      [java] 	at org.apache.tools.ant.helper.DefaultExecutor.executeTargets(DefaultExecutor.java:41)
>      [java] 	at org.apache.tools.ant.Project.executeTargets(Project.java:1251)
>      [java] 	at org.apache.tools.ant.Main.runBuild(Main.java:834)
>      [java] 	at org.apache.tools.ant.Main.startAnt(Main.java:223)
>      [java] 	at org.apache.tools.ant.launch.Launcher.run(Launcher.java:284)
>      [java] 	at org.apache.tools.ant.launch.Launcher.main(Launcher.java:101)
>      [java] Caused by: java.lang.IllegalAccessException: Class org.apache.tools.ant.taskdefs.ExecuteJava can not access a member of class Yylex with modifiers "public static"
>      [java] 	at sun.reflect.Reflection.ensureMemberAccess(Reflection.java:102)
>      [java] 	at java.lang.reflect.AccessibleObject.slowCheckMemberAccess(AccessibleObject.java:296)
>      [java] 	at java.lang.reflect.AccessibleObject.checkAccess(AccessibleObject.java:288)
>      [java] 	at java.lang.reflect.Method.invoke(Method.java:491)
>      [java] 	at org.apache.tools.ant.taskdefs.ExecuteJava.run(ExecuteJava.java:218)
>      [java] 	at org.apache.tools.ant.taskdefs.ExecuteJava.execute(ExecuteJava.java:155)
>      [java] 	... 21 more
>      [java] Java Result: -1
>

### PR DESCRIPTION
**Pull request by lsf37**
* generated class needs to be public to be called from ant (fixes #429)
 * update good test output
 * make ant return failure when junit test fails (#431)
 * refer to lib/ instead of target/ because lib/ is included in release

(Still needs updates to the other examples for #431 to be fixed completely)

### CODE REVIEW

File: jflex/pom.xml [hunk 1]
```diff
@@ -191,6 +191,29 @@
...
+        <artifactId>maven-resources-plugin</artifactId>
+        <executions>
+          <execution>
+            <id>copy-jar-to-lib</id>
+            <phase>install</phase>
+            <goals>
+              <goal>copy-resources</goal>
+            </goals>
+            <configuration>
+              <outputDirectory>${basedir}/lib</outputDirectory>
```
Comment (regisd) [inline jflex/pom.xml]: Why do we need to copy the uberjar in the lib directory? For Maven users, this seems really unnecessary.

I do remember doing something similar for running ant examples in
https://github.com/jflex-de/jflex/blob/8657dfb65629dd29252884e92ea5407404b7c8cf/scripts/test-examples.sh#L32

The difficulty, I guess, is to make ant work both from source code (where the uberjar is in target) and from the release zip (where the uberjar is in lib)

File: jflex/examples/simple/src/main/jflex/simple.flex [hunk 2]
```diff
@@ -1,11 +1,13 @@
-/* this is the scanner example from the JLex website 
+/* this is the scanner example from the JLex website
    (with small modifications to make it more readable) */
 
 %%
 
+%public
```
Comment (regisd) [inline jflex/examples/simple/src/main/jflex/simple.flex]: I also don't understand why the class needs to be public.

File: jflex/pom.xml (hunk 1 above)
Comment (lsf37) [reply jflex/pom.xml]: It’s mostly for being able to run the bin/jflex scripts during dev, eg for testing Makefiles and others the same way one would in the distribution after release. For maven it is indeed unnecessary, but we want people to be able to use other build systems after release. Having the jar in lib up-to-date in dev just makes that a bit easier.

File: jflex/examples/simple/src/main/jflex/simple.flex (hunk 2 above)
Comment (lsf37) [reply jflex/examples/simple/src/main/jflex/simple.flex]: Restriction of the ant run task. Annoying, but not much we can do about it I’d we don’t want to make the example more complicated.

File: jflex/pom.xml (hunk 1 above)
Comment (lsf37) [reply jflex/pom.xml]: (my long-winded way of saying yes, you're right :-))

**Pull request merged**